roles/rsyslog/tasks/set_vars.yml

```diff
@@ -16,3 +16,19 @@
     - "{{ role_path }}/vars/{{ ansible_facts['distribution'] }}_\
       {{ ansible_facts['distribution_version'] }}.yml"
   when: item is file
+
+- name: "Include {{ role_path }}/roles/rsyslog/vars/main.yml"
+  include_vars: "{{ item }}"
+  loop:
+    - "{{ role_path }}/vars/main.yml"
+    - "{{ role_path }}/vars/inputs/basics/main.yml"
+    - "{{ role_path }}/vars/inputs/files/main.yml"
+    - "{{ role_path }}/vars/inputs/ovirt/main.yml"
+    - "{{ role_path }}/vars/inputs/relp/main.yml"
+    - "{{ role_path }}/vars/inputs/remote/main.yml"
+    - "{{ role_path }}/vars/outputs/elasticsearch/main.yml"
+    - "{{ role_path }}/vars/outputs/files/main.yml"
+    - "{{ role_path }}/vars/outputs/forwards/main.yml"
+    - "{{ role_path }}/vars/outputs/relp/main.yml"
+    - "{{ role_path }}/vars/outputs/remote_files/main.yml"
```

Comment:
You could use lookup fileglob here like this:
```yaml
- name: Include rsyslog vars files
  include_vars: "{{ item }}"
  loop: "{{ ['{{ myrole_path }}/vars/main.yml'] +
            lookup('lines', '/bin/ls -1 {{ myrole_path }}/vars/*/*/main.yml').split(',') }}"
```
then, you could install the packages defined in the vars from these files like this:
```
- name: Install rsyslog packages for tests
  package:
    name: "{{ lookup('vars', *lookup('varnames', '^__rsyslog_.*packages$', wantlist=True)) | flatten | unique | sort }}"
    state: present
```
The `varnames` lookup files all vars matching `^__rsyslog_.*packages$` which is the pattern used to define the packages in the rsyslog/vars/*/*/main.yml files.  Then, the `vars` lookup gets the values of those named variables.

Reply:
Thank you, @richm! Your code is much cleaner. I replaced the hardcoded version with your suggestion.